pkg/service/elastic_stack_service.go

```diff
@@ -0,0 +1,188 @@
+	return nil
+}
+
+func NewKibana(kibanaSpec spec.KibanaSpec) *Kibana {
+	return &Kibana{
+		spec:          kibanaSpec,
+		ContainerName: spec.KibanaDefaultContainerName,
+	}
+}
+
+func (k *Kibana) GetServiceName() string {
+	return "kibana"
+}
+
+func (k *Kibana) Display() map[string]utils.DisplayedComponent {
+	cfgDir := k.spec.RemoteCfgPath
+	kibana := utils.DisplayedComponent{
+		Name:          "Kibana",
+		Host:          k.spec.Host,
+		Ports:         strconv.Itoa(k.spec.Port),
+		ContainerName: k.ContainerName,
+		Image:         k.spec.Image,
+		Paths:         strings.Join([]string{cfgDir}, ","),
+	}
+	return map[string]utils.DisplayedComponent{"kibana": kibana}
+}
+
+func (k *Kibana) InitEnv(globalCtx *GlobalCtx) *executor.ExecuteCtx {
+	cfgDir := k.spec.RemoteCfgPath
+	args := append([]string{}, "sudo mkdir -p", cfgDir)
+	return &executor.ExecuteCtx{Target: k.spec.Host, Cmd: strings.Join(args, " ")}
+}
+
+func (k *Kibana) Deploy(globalCtx *GlobalCtx) *executor.ExecuteCtx {
+	mountPoints := []spec.MountPoints{}
+	args := spec.GetDockerExecCmd(globalCtx.containerCfg, k.spec.ContainerCfg, k.ContainerName, true, mountPoints...)
+	args = append(args, k.spec.Image)
+	return &executor.ExecuteCtx{Target: k.spec.Host, Cmd: strings.Join(args, " ")}
+}
+
+func (k *Kibana) Remove(globalCtx *GlobalCtx) *executor.ExecuteCtx {
+	args := []string{"docker rm -f", k.ContainerName}
+	args = append(args, "&&", "sudo rm -rf", k.spec.RemoteCfgPath)
+	return &executor.ExecuteCtx{Target: k.spec.Host, Cmd: strings.Join(args, " ")}
+}
+
+func (k *Kibana) SyncConfig(globalCtx *GlobalCtx) *executor.TransferCtx {
+	return nil
+}
+
+func NewFilebeat(fbSpec spec.FilebeatSpec) *Filebeat {
+	return &Filebeat{
+		spec:          fbSpec,
+		ContainerName: spec.FilebeatDefaultContainerName,
+	}
+}
+func (fb *Filebeat) GetServiceName() string {
+	return "filebeat"
+}
+
+func (fb *Filebeat) Display() map[string]utils.DisplayedComponent {
+	cfgDir := fb.spec.RemoteCfgPath
+	kibana := utils.DisplayedComponent{
+		Name:          "Filebeat",
+		Host:          fb.spec.Host,
+		Ports:         "",
+		ContainerName: fb.ContainerName,
+		Image:         fb.spec.Image,
+		Paths:         strings.Join([]string{cfgDir}, ","),
+	}
+	return map[string]utils.DisplayedComponent{"filebeat": kibana}
+}
+
+func (fb *Filebeat) InitEnv(globalCtx *GlobalCtx) *executor.ExecuteCtx {
+	cfgDir := fb.spec.RemoteCfgPath
+	args := append([]string{}, "sudo mkdir -p", cfgDir)
+	return &executor.ExecuteCtx{Target: fb.spec.Host, Cmd: strings.Join(args, " ")}
+}
+
+func (fb *Filebeat) Deploy(globalCtx *GlobalCtx) *executor.ExecuteCtx {
+	mountPoints := []spec.MountPoints{
+		{"/var/lib/docker", "/var/lib/docker:ro"},
+		{"/var/run/docker.sock", "/var/run/docker.sock"},
+	}
+	if fb.spec.LocalCfgPath != "" {
+		mountPoints = append(mountPoints, spec.MountPoints{Local: fb.spec.LocalCfgPath, Remote: "/usr/share/filebeat/filebeat.yml"})
```

Comment:
mount a directory to a specific file is wrong

Reply:
what if the local config path is required as a single YAML file? the `usr/share/filebeat` path contains many other configs so it can not be written as bind to `usr/share/filebeat`

Reply:
ok, just leave `LocalCfgPath` as a single config file path now